compiler/rustc_hir_analysis/src/coherence/mod.rs

```diff
@@ -49,6 +50,18 @@ fn enforce_trait_manually_implementable(
 ) -> Result<(), ErrorGuaranteed> {
     let impl_header_span = tcx.def_span(impl_def_id);
 
+    if tcx.lang_items().freeze_trait() == Some(trait_def_id) {
+        if !tcx.features().freeze_impls {
+            feature_err(
+                &tcx.sess,
+                sym::freeze_impls,
+                tcx.def_span(impl_def_id),
+                "`Freeze` is automatically implemented by the compiler and should not be implemented manually",
```

Comment:
In the interest of consistency, we might might to make this error message more similar to the one received if you try to manually implement `Sized`:
```
error[E0322]: explicit impls for the `Sized` trait are not permitted
 --> src/lib.rs:4:1
  |
4 | impl Sized for Foo {}
  | ^^^^^^^^^^^^^^^^^^ impl of `Sized` not allowed

For more information about this error, try `rustc --explain E0322`.
```